src/SFA.DAS.ApplyService.Web/Infrastructure/Services/RefreshTrusteesService.cs

```diff
@@ -0,0 +1,138 @@
+            }
+
+            var currentStatusesForUkrpn = await _applicationApiClient.GetExistingApplicationStatus(ukprn);
+            var isStatusInProgress = currentStatusesForUkrpn.Any(x => x.Status==ApplicationStatus.InProgress);
+
+            if (!isStatusInProgress)
+            {
+                _logger.LogInformation($"RefreshTrusteesService: Refresh failure for applicationId {applicationId} as status '{ApplicationStatus.InProgress}' expected.");
+                return new RefreshTrusteesResult { CharityDetailsNotFound = true, CharityNumber = charityNumber };
+            }
+
+            var charityDetails = await _outerApiClient.GetCharityDetails(charityNumberValue);
+
+            if (charityDetails == null || !charityDetails.IsActivelyTrading || charityDetails.Trustees == null || !charityDetails.Trustees.Any())
+            {
+                _logger.LogInformation($"RefreshTrusteesService:  Failure for applicationId {applicationId}  to retrieve trustee details from charity '{charityNumberValue}'");
+                return new RefreshTrusteesResult { CharityDetailsNotFound = true, CharityNumber = charityNumber };
+            }
+
+            _logger.LogInformation($"RefreshTrusteesService: updating organisation trustees applicationId {applicationId}");
+
+            organisation.OrganisationDetails.CharityCommissionDetails.Trustees = charityDetails.Trustees.Select(trusteeValue => new Trustee { Name = trusteeValue.Name, Id = trusteeValue.Id.ToString() }).ToList();
+            organisation.UpdatedBy = userId.ToString();
+
+            try
+            {
+                var success = await _retryPolicy.ExecuteAsync(
+                    context => _organisationApiClient.UpdateTrustees(ukprn, charityDetails.Trustees?.ToList(), userId), new Context());
+
+                if (!success)
+                {
+                    throw new InvalidOperationException($"RefreshTrusteesService for Application {applicationId} update trustees failed");
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new InvalidOperationException($"RefreshTrusteesService for Application {applicationId} update trustees failed: [{ex.Message}]");
+
+            }
+
+            var applicationDetails = new Domain.Roatp.ApplicationDetails { CharitySummary = Mapper.Map<CharityCommissionSummary>(charityDetails) };
+            var trusteesAnswers = RoatpPreambleQuestionBuilder.CreateCharityCommissionWhosInControlQuestions(applicationDetails);
+
+            _logger.LogInformation($"RefreshTrusteesService: resetting page answers for charities for page 80, 86 and section 3.4, applicationId {applicationId}");
+            var resetSection1_3_80 = _qnaApiClient.ResetPageAnswersBySequenceAndSectionNumber(applicationId, RoatpWorkflowSequenceIds.YourOrganisation, RoatpWorkflowSectionIds.YourOrganisation.WhosInControl, RoatpWorkflowPageIds.WhosInControl.CharityCommissionTrustees);
```

Comment:
Why no exception handling during reset requests?

Reply:
I modelled it on the work done in company, but looking at it I realise that has a big try-catch.  Handling now added.